### PR TITLE
JAMES-2318 Expose Cassandra PoolingOptions::setMaxQueueSize

### PR DESCRIPTION
**Pull request by matzepan**
Exposed the pooling option for cassandra, updated the doc, see [JAMES-2318](https://issues.apache.org/jira/browse/JAMES-2318)

### CODE REVIEW

File: src/site/xdoc/server/config-cassandra.xml [hunk 1]
```diff
@@ -76,22 +76,27 @@
...
 
         <dt><strong>cassandra.pooling.timeout</strong></dt>
         <dd>Optional.
-          If specified defines the Cassandra <a href="http://docs.datastax.com/en/developer/java-driver/2.1/manual/pooling/">Specify timeout while not locationg pool</a>. Should be higher than sockets timeout.</dd>
+          If specified defines the Cassandra <a href="https://docs.datastax.com/en/developer/java-driver/3.5/manual/pooling/#acquisition-queue">timeout for waiting in the pool queue</a>. Should be higher than sockets timeout.</dd>
 
         <dt><strong>cassandra.pooling.heartbeat.timeout</strong></dt>
         <dd>Optional.
-          If specified defines the Cassandra <a href="http://docs.datastax.com/en/developer/java-driver/2.1/manual/pooling/">Specify heartbeat timeout</a></dd>
+          If specified defines the Cassandra <a href="https://docs.datastax.com/en/developer/java-driver/3.5/manual/pooling/#heartbeat">heartbeat timeout</a>.</dd>
```
Comment (chibenwa) [inline src/site/xdoc/server/config-cassandra.xml]: Can you specify that the default is 256?

Comment (trantienduchn) [reply src/site/xdoc/server/config-cassandra.xml]: It should be 30 for **heartbreak** timeout

File: src/site/xdoc/server/config-cassandra.xml [hunk 2]
```diff
@@ -76,22 +76,27 @@
...
+
+        <dt><strong>cassandra.pooling.max.queue.size</strong></dt>
+        <dd>Optional.
```
Comment (trantienduchn) [inline src/site/xdoc/server/config-cassandra.xml]: and specify 256 for **queue size** as the default

File: src/site/xdoc/server/config-cassandra.xml [hunk 3]
```diff
@@ -76,22 +76,27 @@
         # cassandra.pooling.timeout=5000
         ## In seconds.
         # cassandra.pooling.heartbeat.timeout=30
+        # cassandra.pooling.max.queue.size=256
 
         <dt><strong>cassandra.pooling.local.max.connections</strong></dt>
         <dd>Optional.
-          If specified defines the Cassandra <a href="http://docs.datastax.com/en/developer/java-driver/2.1/manual/pooling/">maximum number of connections to hosts</a> (remote and local)</dd>
+          If specified defines the Cassandra <a href="https://docs.datastax.com/en/developer/java-driver/3.5/manual/pooling/">maximum number of connections to hosts</a> (remote and local).</dd>
```
Comment (trantienduchn) [inline src/site/xdoc/server/config-cassandra.xml]: will more precise when adding `#pool-size` at the end of the url (https://docs.datastax.com/en/developer/java-driver/3.5/manual/pooling/#pool-size) 👍

File: src/site/xdoc/server/config-cassandra.xml [hunk 4]
```diff
@@ -76,22 +76,27 @@
...
 
         <dt><strong>cassandra.pooling.local.max.requests</strong></dt>
         <dd>Optional.
-          If specified defines the Cassandra <a href="http://docs.datastax.com/en/developer/java-driver/2.1/manual/pooling/">maximum number of concurrent requests per connection</a></dd>
+          If specified defines the Cassandra <a href="https://docs.datastax.com/en/developer/java-driver/3.5/manual/pooling/">maximum number of concurrent requests per connection</a>.</dd>
```
Comment (trantienduchn) [inline src/site/xdoc/server/config-cassandra.xml]: will more precise when adding `#pool-size` at the end of the url (https://docs.datastax.com/en/developer/java-driver/3.5/manual/pooling/#pool-size) 👍

File: src/site/xdoc/server/config-cassandra.xml [hunk 5]
```diff
@@ -76,22 +76,27 @@
...
 
         <dt><strong>cassandra.pooling.timeout</strong></dt>
         <dd>Optional.
-          If specified defines the Cassandra <a href="http://docs.datastax.com/en/developer/java-driver/2.1/manual/pooling/">Specify timeout while not locationg pool</a>. Should be higher than sockets timeout.</dd>
+          If specified defines the Cassandra <a href="https://docs.datastax.com/en/developer/java-driver/3.5/manual/pooling/#acquisition-queue">timeout for waiting in the pool queue</a>. Should be higher than sockets timeout.</dd>
```
Comment (trantienduchn) [inline src/site/xdoc/server/config-cassandra.xml]: could you specify default pooling timeout value?

**Comment by chibenwa** [comment]
I am relaunching the CI build on your pull request

test this please

**Comment by chibenwa** [comment]
Hi,

This pull request has just been merged.

Thank you again for your dedication toward the James project. If you need help, want to [contribute](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20newbie%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC) [documentation](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20documentation%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), new [features](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20feature%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), tests or [bugfix](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20easyfix%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), please don't hesitate to reach us.

This changeset will be available in James 3.2.0 release.

Thanks again,

Cheers,

Benoit

**Pull request closed without merging**